File: java/headFirstDesignPatterns/chap3/starbuzz/src/br/com/starbuzz/drinks/Milk.java
package br.com.starbuzz.drinks;

public class Milk extends CondimentDecorator {

    Beverage beverage;
    
    public Milk(Beverage beverage) {
        this.beverage = beverage;
    }
    
    @Override
    public String getDescription() {
        return this.beverage.getDescription()+", Milk";
    }

    @Override
    public double cost() {
        return .10 + this.beverage.cost();
    }

}

File: c_cpp/headFirst_designPatterns/chap3/starbuzz/src/options/Milk.cpp
/*
 * Milk.cpp
 *
 *  Created on: 24/01/2015
 *      Author: jcarlos
 */

#include "Milk.hpp"

namespace starbuzz {
namespace options {

Milk::Milk(Beverage* beverage) {
    this->beverage = beverage;
}

Milk::~Milk() {
    delete this->beverage;
}

const char* Milk::getDescription(){
    string str = this->beverage->getDescription();
    str += ", Milk";
    return str.c_str();
}

double Milk::getCost(){
    return this->beverage->getCost() + 0.10;
}

} /* namespace options */
} /* namespace starbuzz */
